data/hai/hai reveal 4 middle.txt

```diff
@@ -1498,7 +1498,56 @@ mission "Hai Reveal [A11-B] Permanent Fleets"
 
 
 
-# <insert text only mission(s?) that provides news updates on the election if you land anywhere en route>
+mission "Hai Reveal [A11-C] Outgoing Election Update 1"
+	landing
+	source
+		government "Hai"
+		not planet "Hai-home"
+		attributes "spaceport"
+	to offer
+		has "Hai Reveal [A11] Sound the Alarm: offered"
+	on offer
+		conversation
+			`The spaceport appears to be operating on a skeleton staff with an increased security presence. Plenty of ships are still coming and going, but they seem to be here strictly on business. The usual milling of tourists, haggling of merchants, and perusal of jobs all appear to have been suspended in favor of efficient boarding and disembarking of Hai with places to go and goods with places to be. The displays in the main thoroughfare have been switched over to wall-to-wall election coverage and the rest of the non-essential sections of the spaceport have been closed.`
```

Comment:
This is a landing mission, not a spaceport mission, so you are not actually in the spaceport during this mission.

Reply:
I suggest you make this a spaceport mission, and give it the "major" trait, so no other spaceport missions are offered. Just make sure it doesn't trigger on Mountaintop. If there are several other missions like it, then the player's spaceport missions in Hai space are blocked until the election (A12) finishes, and they can keep getting election updates.

Reply:
This apparently is one of *many* places in "hai reveal 1 intro.txt" where the player is not in the spaceport, but is told they're in the spaceport. I've opened an issue #7804 for that and noted that this PR adds another such bug.

Reply:
> This apparently is one of _many_ places in "hai reveal 1 intro.txt" where the player is not in the spaceport, but is told they're in the spaceport. 

A player must always pass *through* the spaceport to go anywhere. It's the port. You come in on a ship.
`spaceport` missions are for when the player deviates from whichever thoroughfare they must traverse to get from their ship to somewhere else in order to actually explore or spend time within the spaceport itself, rather than simply taking the most direct route through it.

But you *always* go through the spaceport. You've landed a ship, it's unavoidable by definition.

Reply:
The player has deviated from the thoroughfare, to eat at a restaurant and ask questions of many Hai. That is the sort of thing you do when looking for spaceport missions. You are exploring and spending time within the spaceport itself. By your own definition, this is a spaceport mission.

Reply:
The food stalls have been moved into the thoroughfare because the rest of the spaceport is *closed*. That's established in the setup.